feature-libs/cart/saved-cart/components/saved-cart-form-dialog/saved-cart-form-dialog.component.ts

```diff
@@ -151,16 +155,25 @@ export class SavedCartFormDialogComponent implements OnInit, OnDestroy {
     }
   }
 
-  protected build(): void {
-    this.form = new FormGroup({
-      name: new FormControl('', [
+  protected build(cart?: Cart) {
+    const form = new FormGroup({});
+    form.setControl(
+      'name',
+      new FormControl('', [
         Validators.required,
         Validators.maxLength(this.nameMaxLength),
-      ]),
-      description: new FormControl('', [
-        Validators.maxLength(this.descriptionMaxLength),
-      ]),
-    });
+      ])
+    );
+    form.setControl(
+      'description',
+      new FormControl('', [Validators.maxLength(this.descriptionMaxLength)])
+    );
+    this.form = form;
+    this.patchData(cart);
+  }
+
+  protected patchData(item?: any): void {
+    this.form.patchValue({ ...item });
```

Comment:
if the name and description exist already, then the values will be populated in the cart page's dialog as well. You can test this by restoring a cart with a name and description set

Should we not display them when we're saving the cart?
I don't know the answer to be honest, but I think it shouldn't be there.

Reply:
Bill said we can leave the prefill form values on the save cart form (from cart page)